remove recreating page index on migration, later to add as needed

ikreymer opened the pull request:
Don't need it for now, and this will now be slow due to amount of pages. Can readd in future migrations if we need it..

ikreymer committed:
remove recreating page index on migration, later to add as needed

tw4l commented on backend/btrixcloud/db.py:
```diff
@@ -96,7 +95,7 @@ async def update_and_prepare_db(
     await ping_db(mdb)
     print("Database setup started", flush=True)
     if await run_db_migrations(
-        mdb, user_manager, page_ops, org_ops, background_job_ops, coll_ops
```
Quite a few of the migrations use `page_ops` so we can't drop it here, we should only drop it from `craete_indexes` below.

tw4l committed:
Add page_ops back to update_and_prepare_db

tw4l commented on backend/btrixcloud/db.py:
```diff
@@ -216,7 +215,6 @@ async def create_indexes(
     await coll_ops.init_index()
     await invite_ops.init_index()
     await user_manager.init_index()
-    await page_ops.init_index()
```
Won't this mean that new deployments won't have page indices created? I think we might want to check if it already exists rather than just removing the `init_index()` entirely

tw4l replied:
We also haven't yet removed the code that will drop the index, which at this point would force us to recreate it

tw4l replied:
It turns out `create_index` is idempotent, so I've reversed the logic in this PR. Rather than removing the call to create page indices, we now remove the logic that drops the page indices after successful migration.

tw4l committed:
Add page index creation back, as is idempotent

Page indices will only be created if they don't already exist, so
rather than removing this call we should instead make sure they
are not unnecessarily dropped.

tw4l reviewed (approved):
This should be all good now

ikreymer commented:
Thanks for the fix!
